lib/dotiw/methods.rb

```diff
@@ -94,16 +123,43 @@ def _display_time_in_words(hash, options = {})
         end
       end
 
-      output = []
-      I18n.with_options locale: options[:locale], scope: i18n_scope do |locale|
-        output = hash.map { |key, value| locale.t(key, count: value) }
-      end
-
       options.delete(:except)
       options.delete(:only)
+
       highest_measures = options.delete(:highest_measures)
       highest_measures = 1 if options.delete(:highest_measure_only)
-      output = output[0...highest_measures] if highest_measures
+      highest_measures = { count: highest_measures } if highest_measures.is_a?(Integer)
+      highest_measures = highest_measures.reverse_merge(count: 1, remainder: :floor) if highest_measures
+
+      if highest_measures
+        high_entries, low_entries = hash.to_a.partition.with_index { |_, index| index < highest_measures[:count] }
+        hash = high_entries.to_h
+        discarded_hash.merge! low_entries.to_h
+
+        smallest_unit_index = DOTIW::TimeHash::TIME_FRACTIONS.index high_entries.last[0]
+        smallest_unit = DOTIW::TimeHash::TIME_FRACTIONS[smallest_unit_index]
+        case highest_measures[:remainder]
+        when :ceiling
+          # We already filtered out zeroes, so non-empty also means non-zero.
+          if !discarded_hash.empty?
+            hash[smallest_unit] += 1
+            _accumulate_upwards! hash, smallest_unit_index
+          end
+        when :round
+          if smallest_unit_index > 0
```

Comment:
All the index noodling in this `if highest_measures` block is annoying to follow, but this `if` in particular is kinda gnarly. Thoughts?

Reply:
It is. I'd just extract in a private method to make it more readable. Also when does this happen?

Reply:
I pulled it out and rearranged it slightly so it's hopefully a little clearer. I also added a code comment to answer your question, since it's very much not obvious.